src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -149,7 +148,7 @@ class HierarchicalNSW : public VecsimBaseObject {
     void SwapLastIdWithDeletedId(tableint element_internal_id, tableint last_element_internal_id);
 
 public:
-    HierarchicalNSW(SpaceInterface<dist_t> *s, size_t max_elements,
+    HierarchicalNSW(VecSimType vecType, VecSimMetric metric, size_t dim, size_t max_elements,
```

Comment:
Why not send HNSWParams struct instead of long parameters list?

Reply:
good idea